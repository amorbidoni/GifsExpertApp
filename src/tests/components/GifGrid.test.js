import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifGridApp from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("pruebas en el componente GifGrid", () => {
  const category = "los simpson";
  test("corroborar que recibo correctamente el componente", () => {
    useFetchGifs.mockReturnValue({ data: [], loading: true });

    const wrapper = shallow(<GifGridApp category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar items cuando se cargan imagenes con useFetchGif", () => {
    //
    const gifs = [
      {
        id: "asd",
        title: "title",
        url: "https://localhost/4200",
      },
      {
        id: "123",
        title: "title",
        url: "https://localhost/4200",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGridApp category={category} />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItemApp").length).toBe(gifs.length);
  });
});
