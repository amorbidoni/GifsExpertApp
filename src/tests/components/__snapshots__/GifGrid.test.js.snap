// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente GifGrid corroborar que recibo correctamente el componente 1`] = `
<Fragment>
  <h1
    className=" animate__animated animate__fadeInUp"
  >
    los simpson
  </h1>
  <p>
    Cargando...
  </p>
  <div
    className="card__grid  animate__animated animate__fadeInUp"
  />
</Fragment>
`;

exports[`pruebas en el componente GifGrid debe mostrar items cuando se cargan imagenes con useFetchGif 1`] = `
<Fragment>
  <h1
    className=" animate__animated animate__fadeInUp"
  >
    los simpson
  </h1>
  <div
    className="card__grid  animate__animated animate__fadeInUp"
  >
    <GifGridItemApp
      id="asd"
      key="asd"
      title="title"
      url="https://localhost/4200"
    />
    <GifGridItemApp
      id="123"
      key="123"
      title="title"
      url="https://localhost/4200"
    />
  </div>
</Fragment>
`;
